website/thaliedje/services.py

```diff
@@ -20,9 +22,64 @@ def user_is_blacklisted(user):
     return ThaliedjeBlacklistedUser.objects.filter(user=user).exists()
 
 
-def has_album_playlist_request_permission(user, player):
-    """Check if a user has album and playlist request permissions."""
-    return user.has_perm("thaliedje.can_request_playlists_and_albums", player)
+def active_thaliedje_control_event(player):
+    """
+    Get the active ThaliedjeControlEvent for a Player.
+
+    :param player: the player
+    :return: the active ThaliedjeControlEvent or None
+    """
+    qs = ThaliedjeControlEvent.objects.filter(player=player.id, active=True)
+    if qs.exists():
+        return qs.first()
+    return None
+
+
+def can_request_song(user, player):
+    """Check if a user can request a song."""
+    control_event = active_thaliedje_control_event(player)
+    if control_event is not None:
+        if control_event.respect_blacklist and user_is_blacklisted(user):
+            return False
```

Comment:
Maybe check this in the can_request_song function on the control event as well

Reply:
What do you mean exactly?

Reply:
Wait I understand. Its not possible because of circular dependencies